src/main/java/edu/cornell/kfs/paymentworks/service/impl/PaymentWorksKfsServiceImpl.java

```diff
@@ -64,387 +59,312 @@
+		return vendorMaintDoc;
+	}
+
+	protected boolean processUnlockedVendorMaintenanceDocument(StringBuffer documentNumbers, MaintenanceDocument vendorMaintDoc) throws WorkflowException {
+		boolean isValid = true;
+		boolean isRouted = false;
+		try {
+			vendorMaintDoc.validateBusinessRules(new RouteDocumentEvent(vendorMaintDoc));
+		} catch (ValidationException ve) {
+			LOG.error("Failed to route Vendor Edit document due to business rule error(s): "
+					+ new PaymentWorksUtil().getAutoPopulatingErrorMessages(GlobalVariables.getMessageMap().getErrorMessages()));
+			isValid = false;
+		}
+
+		if (isValid) {
+			// addNoteForNewVendorResponsesToVendor(vendorMaintDoc.getNewMaintainableObject());
+			documentService.routeDocument(vendorMaintDoc, null, null);
+			isRouted = true;
+			documentNumbers.append(vendorMaintDoc.getDocumentNumber() + ",");
 		}
 		return isRouted;
 	}
 
 	@Override
 	public boolean directVendorEdit(PaymentWorksVendor paymentWorksVendor) {
-
 		boolean success = true;
 		PaymentWorksVendorUpdateConversionUtil paymentWorksVendorUpdateConversionUtil = new PaymentWorksVendorUpdateConversionUtil();
-
 		String vendorNumberList = paymentWorksVendor.getVendorNumberList();
 		List<String> vendorNumbers = Arrays.asList(vendorNumberList.split("\\s*,\\s*"));
 
-		// loop through vendors
 		for (String vendorNumber : vendorNumbers) {
-
-			// retrieve vendor from kfs
 			VendorDetail vendorDetail = vendorService.getVendorDetail(vendorNumber);
-
 			if (ObjectUtils.isNotNull(vendorDetail)) {
 				if (!paymentWorksVendorUpdateConversionUtil.duplicateFieldsOnVendor(vendorDetail, paymentWorksVendor)) {
-					// set the necessary fields
-					vendorDetail = paymentWorksVendorUpdateConversionUtil.createVendorDetailForEdit(vendorDetail, null,
-							paymentWorksVendor);
-
-					// save to DB, no document
+					vendorDetail = paymentWorksVendorUpdateConversionUtil.createVendorDetailForEdit(vendorDetail, null, paymentWorksVendor);
 					try {
 						businessObjectService.save(vendorDetail);
 					} catch (Exception e) {
-						GlobalVariables.getMessageMap().putError("vendorDetail", KFSKeyConstants.ERROR_CUSTOM,
-								e.getCause().getMessage());
+						GlobalVariables.getMessageMap().putError("vendorDetail", KFSKeyConstants.ERROR_CUSTOM, e.getCause().getMessage());
 						success = false;
 						break;
 					}
 				} else {
-					GlobalVariables.getMessageMap().putError("vendorDetail", KFSKeyConstants.ERROR_CUSTOM,
-							"Duplicate vendor update detected");
+					GlobalVariables.getMessageMap().putError("vendorDetail", KFSKeyConstants.ERROR_CUSTOM, "Duplicate vendor update detected");
 					success = false;
 				}
 			}
 		}
-
 		return success;
 	}
-
+	
 	@Override
 	public boolean directAchEdit(PaymentWorksVendorUpdatesDTO vendorUpdate, String vendorNumberList) {
-
 		boolean isSuccessful = false;
-
 		String routingNumber = getRoutingNumber(vendorUpdate.getField_changes().getField_changes());
```

Comment:
The DTO objects in xmlObjects package all use under scores in the field names. This allows the mapping of POJO variables to the payment works XML without the need for explicit annotations. I would prefer not to change that.